docs/user/reporting/network-policy.asciidoc

```diff
@@ -0,0 +1,74 @@
+[role="xpack"]
+[[reporting-network-policy]]
+=== Restricting requests with a Reporting Network Policy
+
+When {report-features} generates PDF reports, it uses the Chromium browser to fully load the Kibana page on the server. This
+potentially involves sending requests to external hosts, for example, a request may go to an external image server for showing a
+field formatted as an image, or to show an image in a Markdown visualization.
+
+If the Chromium browser is requested to send a request that violates the network policy, Reporting will stop processing the page
+before the request goes out, and the report will be marked as a failure. Additional information about the event can be found in
+Kibana's server logs.
+
+A network policy applies not only to outgoing requests, but also incoming responses. That means if a request goes out to an allowed
+host, but is redirected and a response returns from a denied host, the response will be denied, and the report will fail.
```

Comment:
nit: Isn't this still technically an outgoing request? The redirect would involve a response from the allowed host, and then the browser would issue a new request to the denied host -- the network policy would then prevent the connection to the denied host.

The way it's currently written, I could misinterpret this to believe that Chromium would connect to the denied host, but simply ignore the response.

Reply:
Yes, I mistook what is going on in the response handler code that I have become slightly familiar with.

I should have realized the point you made, but now I'm less sure about what the code is doing  :)

Reply:
Yup, that's correct: we do a best effort prior to the request being sent, but if a DNS hijack or something else occurs, this can redirect to internal-only IPs or worse (the `file://` protocol, etc).

This is why we check both the request itself, as well as the response, and if any violate network policies then the session is shutdown immediately (the browser is closed, we log the reason, and the reporting job is failed).

Reply:
Cool, glad to understand it better. 

I think that level of detail might not be required for the docs. I'm just going to wipe out these sentences.